02b_widgets.ipynb

```diff
@@ -5,106 +5,145 @@
+   "id": "8bbdd068",
    "metadata": {},
    "source": [
-    "## Preview the Final Product"
+    "## Building the Widgets Module of our Package\n",
+    "\n",
+    "We will be developing the `widgets.py` module in this notebook, so we start by declaring that module to be our intended export target in `nbdev`. Notice that as we work through this notebook, we will export only those cells necessary to create the `widgets.py` module, leaving other cells which provide examples our testing or development process intact."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "d8a59617-f05b-48fc-afcd-79f534a02e96",
+   "id": "b3026a02-fb27-4620-a6b0-3378e2d37f27",
    "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "application/vnd.jupyter.widget-view+json": {
-       "model_id": "361e3273c8784ac59db7b491fe95cd56",
-       "version_major": 2,
-       "version_minor": 0
-      },
-      "text/plain": [
-       "HBox(children=(VBox(children=(HTML(value='\\n<p><b>Curve Smoothing</b>\\nThis tool is for smoothing and selectin…"
-      ]
-     },
-     "execution_count": null,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
+   "outputs": [],
    "source": [
-    "from key.dashboard.main import main_widget\n",
+    "#| default_exp widgets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41ed38cf-fc0b-4ef5-8c9b-2619f4ab3058",
+   "metadata": {},
+   "source": [
+    "## Preview the Final Product\n",
+    "\n",
+    "We will be converting our previous data workflow into a fully interactive web-based dashboard using Jupyter widgets.  The final product will look something like this:\n",
     "\n",
-    "main_widget"
+    "![Preview Image of Data Dashboard](key/dashboard_screenshot.png)"
    ]
   },
   {
    "cell_type": "markdown",
    "id": "8ad7eb7b-1a62-494d-82bb-bb457ebbf6b2",
    "metadata": {},
    "source": [
-    "### Import ipywidgets\n",
-    "Let's add an import statement for ipywidgets. We will use a lot of widgets, so it's good to import ipywidgets as widgets, or something similiarly short. Add an import statement below and then go to [ipywidgets documentation site](https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html) to get a sense of what kind of widgets there are."
+    "## Reviewing Jupyter Widgets\n",
+    "\n",
+    "Let's add an import statement for Jupyter widgets (aka `ipywidgets`). We will use a lot of widgets, so it's good to `import ipywidgets as widgets`, or something similarly short. \n",
+    "\n",
+    "Add an import statement below:"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 3,
    "id": "e5c6ecde-5850-4d35-903e-6ed258669fe6",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "UsageError: Line magic function `%answer` not found.\n"
+     ]
+    }
+   ],
```

Comment:
Just a friendly reminder that doing a "Restart and Run all cells" before commit makes it easier to review the diff, which I'm sure you know already and I forget frequently as well :)

Reply:
Dang, sorry about that. :)